### PR TITLE
feat(dexes): add edit dex modal

### PR DESCRIPTION
**Pull request by ami**
୧☉□☉୨ THE END IS NEAR

to-do for you to add:
- [x] add delete functionality
- [x] auto fill form with selected dex
- [x] add edit functionality
- [x] add functionality to toggle errors only if those inputs are updated
- [x] only "warn" the user on submit if there are warnings

### CODE REVIEW

**Comment by robinjoseph08** [comment]
@ami updatedddd

File: app/components/dex-edit.jsx [hunk 1]
```diff
@@ -21,79 +28,97 @@ export class DexEdit extends Component {
...
               <i className="fa fa-asterisk" />
             </div>
             <div className="form-group">
-            <div className="form-warning">
-              <div className="tooltip">
-                <i className="fa fa-exclamation-triangle" />
-                <span className="tooltip-text">Any Gen 7 capture information will be lost.</span>
-              </div>
-            </div>
+              <FormWarningComponent message="Any newer generation capture information will be lost." show={gen < dex.generation} />
```
Comment (ami) [inline app/components/dex-edit.jsx]: let's have a warning appear when going Gen 6 -> Gen 7 also, with appropriate wording

**Comment by robinjoseph08** [comment]
@ami updated!!

**Comment by ami** [comment]
LGTMMMM

**Pull request merged**